charts/htsget-rs/values.yaml

```
@@ -0,0 +1,180 @@
+
+nodeSelector: {}
+
+tolerations: []
+
+affinity: {}
+
+# The application's ticket server TLS can be disabled by setting enabled to false.
+# Not to be used in production but handy for testing.
+tls:
+  enabled: false
+  issuer: ""
+  clusterIssuer: ""
+  secretName: ""
+
+c4gh:
+  # Set to true if htsget will use a predefined c4gh key-pair instead
+  # of creating one on-the-fly when handling encrypted data
+  predefined: false
+  # The name for the secret containing c4gh keys.
+  secretName: ""
+
+# Enable data server TLS
+tlsDataServer:
+  enabled: false
+  secretName: ""
+
+# Enable client TLS for the urlstorage backend (experimental feature)
+tlsClient:
+  enabled: false
+  secretName: ""
+
+htsget:
+  # Logging levels and formatting (inherited by rust)
+  rustLog: "info" # debug, trace
+  formattingStyle: "Pretty" # Full, Compact, Pretty, Json
+
+  # These should match the path in the htsget config below.
+  # Path where the sensitive files can be found.
+  c4ghPath: ""
+  tlsPath: "/tls"
+  tlsPathDataServer: ""
+  tlsPathClient: ""
+
+  # If the data server is enabled, use of a local storage backend can be enabled by setting the localPath to a path where the data files will be mounted
+  dataServer:
+    enabled: true
+    localPath: "/data"
+    # Data can be mounted to the local path via an existing persistent volume claim
+    existingClaim: ""
+    # For a nfs volume there is also the option to set the configuration below instead of using a persistent volume claim
+    nfsServer: ""
+    nfsPath: ""
+
+# The htsget-rs server is configured by placing the contents of the toml config file below (keep nindent 2).
+# All of the available configuration options (https://github.com/umccr/htsget-rs/tree/crypt4gh/htsget-config)
+# can be set this way. Paths to sensitive files like TLS certicifaces or c4gh keys need to be set above and match
+# those used below. The names of the files that are referenced below should be the same as the ones in the respective
+# secrets whose names can be set above.
+# Below is an example of a htsget-rs config file utilizing a url_storage.
+configMapData: |-
+  # Information displayed by the /info endpoint
+  name = "htsget.example"
+  version = ""
+  organization_name = "Example Organization"
+  organization_url = "http://example.com"
+  contact_url = "http://example.com"
+  documentation_url = "https://github.com/umccr/htsget-rs"
+
+  # The ticket server configuration
+  ticket_server_addr = "127.0.0.1:8080"
+  ticket_server_cors_allow_origins = "All"
+  # ticket_server_tls.key = "/tls/tls.key"
+  # ticket_server_tls.cert = "/tls/tls.crt"
+
+  # The data server configuration
+  data_server_enabled = true
+  data_server_addr = "127.0.0.1:8081"
+  data_server_cors_allow_origins = "All"
+  data_server_local_path = "/data"
+  # data_server_tls.cert = "/tlsDataServer/tls.crt"
+  # data_server_tls.key = "/tlsDataServer/tls.key"
+
+  [[resolvers]]
+  regex = ".*"
+  substitution_string = "$0"
```

```suggestion
  substitution_string = "$0"
  
  [resolvers.object_type]
  # if set to true, responseUrl must point to an endpoint serving encrypted data
  send_encrypted_to_client = true
  # if c4gh keys are not set, they will be generated on the fly
  # only custom keys without a passphrase are supported
  private_key = "/c4gh/c4gh.sec.pem"
  public_key = "/c4gh/c4gh.pub.pem"

  [resolvers.storage]
  # The url that will be used for the client's url
  response_url = "http://example.com"
  forward_headers = true
  # if set to true, the client will not check the certificate validity
  danger_accept_invalid_certs = true

  [resolvers.storage.endpoints]
  index = "http://example.com"
  file = "http://example.com"
```


Main values file is missing S3 storage example

---

The dataserver is incompatible with the above resolvers configuration that is why they are omitted. Further examples for these configs exist under `.github/integration/scripts` and referenced in the README.